srcCpp/RTIDDSImpl.cxx

```diff
@@ -2224,7 +2209,9 @@ IMessagingWriter *RTIDDSImpl<T>::CreateWriter(const char *topic_name)
         return NULL;
     }
     
-    if (!_UsePositiveAcks) {
+    if (!_UsePositiveAcks
+            && (qos_profile == "ThroughputQos" || qos_profile == "LatencyQos")) {
```

Comment:
You should use strcmp. This is a pointer compare, which may work because the compiler will likely define each literal only once. But that is still not maintainable or a good practice.

Reply:
qos_profile is a std::string, not a char*